### PR TITLE
beamer: config: remove chain from dictionary if reset to default values

### PR DESCRIPTION
**Pull request by fredo**
The same thing was missing for chains. 

Removing chains should result in removing it from the dicitonary.

### CODE REVIEW

**Review by bilbeyt — APPROVED** [review]
Seems good to me, checked if we have any tests regarding Token removal and couldnt find any so this should be also ok

**Comment by istankovic** [comment]
> Seems good to me, checked if we have any tests regarding Token removal and couldnt find any so this should be also ok

Should we not then use this opportunity to add new tests, both for token and chain removal? :smile: 
Especially since it seems like an easy thing to do.

**Comment by fredo** [comment]
> > Seems good to me, checked if we have any tests regarding Token removal and couldnt find any so this should be also ok
> 
> Should we not then use this opportunity to add new tests, both for token and chain removal? 😄 Especially since it seems like an easy thing to do.
> 
> It would also be interesting to see test coverage now for modules under `beamer.config`.

There you go

File: beamer/tests/config/test_read.py [hunk 1]
```diff
@@ -61,6 +71,14 @@ def test_config_read_request_manager(tmp_path, token, deployer):
     assert chain_config.transfer_cost == 7
     assert chain_config.target_weight_ppm == 8
 
+    # check chain removal
+    request_manager.updateChain(123, 0, 0, 0)
+    config = read_config_state(rpc_file, artifact)
+
+    assert config.block > old_block
+    assert symbol not in config.request_manager.chains
+    assert len(config.request_manager.tokens) == 0
```
Comment (istankovic) [inline beamer/tests/config/test_read.py:80]: This should have been
```suggestion
    assert len(config.request_manager.chains) == 0
```
right?

**Review by istankovic — APPROVED** [review]

File: beamer/tests/config/test_read.py (hunk 1 above)
Comment (fredo) [reply beamer/tests/config/test_read.py]: it was late last night... pushed a fix

**Pull request merged**